configure.ac

```
@@ -300,13 +301,9 @@ AC_ARG_ENABLE([frame-pointers],
   [AS_HELP_STRING([--enable-frame-pointers],
     [use frame pointers in runtime and generated code])])
 
-AC_ARG_ENABLE([naked-pointers],
-  [AS_HELP_STRING([--disable-naked-pointers],
-    [do not allow naked pointers])])
-
-AC_ARG_ENABLE([naked-pointers-checker],
-  [AS_HELP_STRING([--enable-naked-pointers-checker],
-    [enable the naked pointers checker])])
+AC_ARG_ENABLE([naked-pointers], [],
+  [AC_MSG_ERROR([Naked pointers are not allowed in OCaml Multicore.])],
+  [AC_DEFINE([NO_NAKED_POINTERS])])
```

This doesn't quite do what's intended (in passing, I see the spacetime one is wrong too, which is embarrassing given that I wrote it!):
```suggestion
AC_ARG_ENABLE([naked-pointers], [],
  [AS_IF([test "x$enableval" != 'xno'],
    [AC_MSG_ERROR([Naked pointers were prohibited in in OCaml 5.00.])])],
  [])
  
AC_ARG_ENABLE([naked-pointers-checker], [],
  [AS_IF([test "x$enableval" != 'xno'],
    [AC_MSG_ERROR([The naked pointers checker was removed in OCaml 5.00.])])],
  [])
```

The `AC_DEFINE` can instead be done this way:

```diff
diff --git a/runtime/caml/m.h.in b/runtime/caml/m.h.in
index 1c3dee1779..359e0b040c 100644
--- a/runtime/caml/m.h.in
+++ b/runtime/caml/m.h.in
@@ -82,9 +82,7 @@

 #undef WITH_FRAME_POINTERS

-#undef NO_NAKED_POINTERS
-
-#undef NAKED_POINTERS_CHECKER
+#define NO_NAKED_POINTERS 1

 #undef WITH_PROFINFO
```
(we should indeed ensure that `m.h` continues to define `NO_NAKED_POINTERS`)

---

Tracking this for post-merge fixing in https://github.com/ocaml/ocaml/issues/10861